components/homme/cmake/CxxVsF90.cmake.in

```diff
@@ -15,29 +15,120 @@ FOREACH (NC_FILE_NAME @NC_OUTPUT_FILES@)
 
   # If cprnc does not return 0, then the test failed
   IF (identical_pos EQUAL -1)
-    MESSAGE (FATAL_ERROR "Test did not succeed. Netcdf outputs '${NC_FILE_NAME}' differ.")
     MESSAGE ("${cprnc_stdout}")
+    MESSAGE (FATAL_ERROR "Test did not succeed. Netcdf outputs '${NC_FILE_NAME}' differ.")
   ENDIF()
 ENDFOREACH()
 
 MESSAGE ("CXX and F90 netcdf outputs match.\n")
 
-# Grep F90 and CXX raw output for all lines containing the string '/dt'.
-# As of today, these are all and only the lines where diagnostics are printed.
-# Then compare the output strings. If they are equal, then the two tests print
-# the same diagnostics
 
-EXECUTE_PROCESS (COMMAND grep ^diagnostics> @F90_DIR@/@F90_TEST_NAME@_1.out
+
+#stdout part: -----------------------------------------------------------------------------
+
+string(FIND @CXX_TEST_NAME@ "preqx" preqxtest)
+#preqxtest = -1 meants the test is not preqx test
+if(preqxtest EQUAL -1)
+
+# USE THIS as an example
```

Comment:
Please remove this block. We can check test output at any time to determine its structure.